sentry/src/main/java/io/sentry/Span.java

```diff
@@ -163,6 +188,30 @@ public void finish(final @Nullable SpanStatus status, final @Nullable SentryDate
 
     this.context.setStatus(status);
     this.timestamp = timestamp == null ? hub.getOptions().getDateProvider().now() : timestamp;
+    if (options.isTrimStart() || options.isTrimEnd()) {
+      @Nullable SentryDate minChildStart = null;
+      @Nullable SentryDate maxChildEnd = null;
+
+      final @NotNull List<Span> children = getChildren();
+      for (final Span child : children) {
+        if (child.getParentSpanId() != null && child.getParentSpanId().equals(getSpanId())) {
```

Comment:
`m`: we don't need this check, because we already do it in `getChildren`, right?